src/core/session/session.go

```diff
@@ -136,7 +136,11 @@ func (rp *Provider) SessionExist(sid string) bool {
 func (rp *Provider) SessionRegenerate(oldsid, sid string) (session.Store, error) {
 	ctx := context.TODO()
 	if !rp.SessionExist(oldsid) {
-		rp.c.Save(ctx, sid, "", time.Duration(rp.maxlifetime))
+		err := rp.c.Save(ctx, sid, "", time.Duration(rp.maxlifetime))
+		if err != nil {
+			log.Errorf("failed to save sid=%s, where oldsid=%s, error: %s", sid, oldsid, err)
+			return nil, err
```

Comment:
don't return early because this change maybe has impact to session because this lib implemented referenced by beego session and follow their behavior.